hive-llap/llap.sh

```diff
@@ -0,0 +1,353 @@
+# be done on all nodes BUT the actual permissions/ownership needs only to be done on the workers. 
+function configure_SSD_caching_worker(){
+    if [[ -n "$HAS_SSD" ]]; then
+        echo "configure ssd hive-site params on workers"
+        mkdir /mnt/1/llap
+        chown hive:hadoop /mnt/1/llap
+        chmod 770 /mnt/1/llap
+
+        bdconfig set_property \
+            --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+            --name 'hive.llap.io.allocator.mmap' --value 'true' \
+            --clobber
+        bdconfig set_property \
+            --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+            --name 'hive.llap.io.allocator.mmap.path' --value '/mnt/1/llap' \
+            --clobber
+        bdconfig set_property \
+            --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+            --name 'hive.llap.io.memory.mode' --value 'cache' \
+            --clobber
+        bdconfig set_property \
+            --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+            --name 'hive.llap.io.use.lrfu' --value 'true' \
+            --clobber
+    fi
+}
+# if the user has added ssd=1 as a metadata option, then we need to configure hive-site.xml for using the ssd as a memory cache extension. 
+function configure_SSD_caching_master(){
+    if [[ -n "$HAS_SSD" ]]; then
+        echo "configure ssd hive-site params on master"
+
+        bdconfig set_property \
+            --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+            --name 'hive.llap.io.allocator.mmap' --value 'true' \
+            --clobber
+        bdconfig set_property \
+            --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+            --name 'hive.llap.io.allocator.mmap.path' --value '/mnt/1/llap' \
+            --clobber
+        bdconfig set_property \
+            --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+            --name 'hive.llap.io.memory.mode' --value 'cache' \
+            --clobber
+        bdconfig set_property \
+            --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+            --name 'hive.llap.io.use.lrfu' --value 'true' \
+            --clobber
+fi
+}
+
+# start LLAP - Master Node
+function start_llap(){
+    if [[ "${HOSTNAME}" == "${LLAP_MASTER_FQDN}" ]]; then
+        echo "starting llap on master node 0..."
+        ."${INIT_ACTIONS_DIR}"/start_llap.sh
+    fi
+}
+
+# main driver function for the script
+function configure_llap(){
+
+    if [[ "${HOSTNAME}" == "${LLAP_MASTER_FQDN}" ]]; then
+        echo "running primary master config...."
+        pre_flight_checks
+        download_init_actions
+        package_tez_lib_uris
+        add_yarn_service_dir
+        configure_yarn_site
+        configure_core_site
+        configure_hive_site
+        configure_SSD_caching_master
+        replace_core_llap_files
+        get_log4j
+        configure_tez_site_xml
+        return 0
+    fi
+
+    if [[ "${ROLE}" == "Worker" ]]; then
```

Comment:
no need to return within the block. You can use if-elif statements here.

Reply:
fixed